internal/metastructure/resource_persister/resource_persister.go

```diff
@@ -426,5 +435,26 @@ func (rp *ResourcePersister) persistTargetUpdate(update *target_update.TargetUpd
 		"operation", update.Operation,
 		"version", version)
 
+	// Trigger discovery for freshly added discoverable targets
+	if target_update.ShouldTriggerDiscovery(update) {
+		rp.Log().Debug("Triggering discovery for newly discoverable target",
+			"label", update.Target.Label,
+			"operation", update.Operation)
+
+		result, err := discovery.TriggerOnce(rp)
```

Comment:
We are passing the `ResourcePersister` actor to a function in the discovery package, which feels a bit unusual. We should have the `ResourcePersister` use the actornames package to find the Discovery actor’s address and send the message directly from this code. Also, the extra message and synchronous handler adds a bunch of complexity, only to show a log message that’s only visible if users are running the agent locally in foreground mode. Can we used the async `Discover { Once: true }` message instead? I understand the goal of this effort is to start discovery right after the first target is defined, which is solid. Handling subsequent targets is a nice addition, but addressing the case where a target is added _exactly_ during discovery is probably overkill.